src/transformers/models/rag/modeling_tf_rag.py

```diff
@@ -0,0 +1,1524 @@
+# coding=utf-8
+# Copyright 2020, The RAG Authors and The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""TFRAG model implementation. (draft version)"""
+
+from dataclasses import dataclass
+from typing import Dict, List, Optional, Tuple
+
+import tensorflow as tf
+from tensorflow import Tensor
+from tensorflow.keras.layers import Dense, LayerNormalization
```

Comment:
Please, replace everywhere by the full name: `tf.Tensor`, `tf.keras.layers.Dense` and `tf.keras.layers.LayerNormalization`